lib/improver/metadata/check_datatypes.py

```diff
@@ -0,0 +1,222 @@
+            if fix:
+                coord.points = coord.points.astype(np.float32)
+            else:
+                raise TypeError(
+                    "64 bit coord points not allowed: {} in {!r}".format(
+                        coord, cube))
+        if (hasattr(coord, "bounds") and coord.bounds is not None and
+                coord.bounds.dtype == np.float64):
+            if fix:
+                coord.bounds = coord.bounds.astype(np.float32)
+            else:
+                raise TypeError(
+                    "64 bit coord bounds not allowed: {} in {!r}".format(
+                        coord, cube))
+
+
+def check_time_coordinate_metadata(cube):
+    """
+    Function to check time coordinates against the expected standard. The
+    standard for time coordinates is due to technical requirements and if
+    violated the data integrity cannot be guaranteed; so if time coordinates
+    are non-conformant an error is raised.
+
+    Args:
+        cubes (iris.cube.Cube):
+            Cube to be checked
+
+    Raises:
+        ValueError: if any the input cube's time coordinates do not conform
+            to the standard datatypes and units
+    """
+    error_string = ''
+    for time_coord in TIME_METADATA.keys():
+        try:
+            coord = cube.coord(time_coord)
+        except CoordinateNotFoundError:
+            continue
+        reqd_unit = TIME_METADATA[time_coord]["unit"]
+        reqd_dtype = TIME_METADATA[time_coord]["dtype"]
+
+        if not _check_units_and_dtype(coord, reqd_unit, reqd_dtype):
+            msg = ('Coordinate {} does not match required '
+                   'standard (units {}, datatype {})\n')
+            error_string += msg.format(
+                coord.name(), reqd_unit, reqd_dtype)
+
+    # if non-compliance was encountered, raise all messages here
+    if error_string:
+        raise ValueError(error_string)
+
+
+def _construct_object_list(cube, coords):
+    """
+    Construct a list of objects to check
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube to be checked
+        coords (list or None):
+            List of coordinate names to check.  If None, adds all
+            coordinates present on the input cube
+
+    Returns:
+        object (list):
+            List containing the original cube and coordinates to check
+    """
+    object_list = []
+    object_list.append(cube)
+    if coords is not None:
+        for coord in coords:
+            try:
+                object_list.append(cube.coord(coord))
+            except CoordinateNotFoundError:
+                pass
+    else:
+        object_list.extend(cube.coords())
+    return object_list
+
+
+def check_datatypes(cube, coords=None):
+    """
+    Function to check the datatypes of cube diagnostics and coordinates
+    against the expected standard.  The default datatype is float32;
+    integer quantities are expected to be 32-bit with the exception of
+    absolute time.
```

Comment:
Again it wouldn't hurt to link to where the standards are set.

Reply:
See above; there is no suitable link for this.

Reply:
Function now does exactly this, without reference to external dictionaries.